Structure_Engine/Modify/SetLocalOrientation.cs

```diff
@@ -44,12 +44,15 @@ public static partial class Modify
         [Output("panel", "The Panel with updated orientation.")]
         public static Panel SetLocalOrientation(this Panel panel, Vector localX)
         {
+            if (panel.IsNull() || localX.IsNull())
+                return null;
```

Comment:
I think returning `null` makes sense based on your fourth bullet point - it essentially breaks your script at a point so you can identify where the issue is instead of lulling you in to a false sense of security. 

On the third point how useful would a `Panel` that is `null` be downstream? Anything you try to do it will fail.

Reply:
> On the third point how useful would a Panel that is null be downstream? Anything you try to do it will fail.

Mainly if the panel is not null, but the vector is, then you would still have the unchanged panel rather than a null.

Reply:
Agreed - think return `null`.

Here I think as we are trying to Modify a valid object with a `Null` input - the actual method should indeed fail. And returning a `null` object makes sense.

I think this is slightly different to the query case [here](https://github.com/BHoM/BHoM_Engine/pull/2504#discussion_r646505189) where we do not what to make the return type _too intelligent!_ It is just a double. So should avoid `NaN`s.

In this case the modify action is indeed invalidating the original panel - and so should return a null object (and in the case of workflows - disrupt downstream)

To me this makes sense 😄 - will be good to summarise the above/agreed conventions in the wiki.
@IsakNaslundBh @peterjamesnugent @FraserGreenroyd